app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsUpdates.kt

```diff
@@ -47,19 +54,27 @@ class SettingsUpdates : PreferenceFragmentCompat() {
         setToolBarScrollFlags()
     }
 
+    private val pathPicker = getChooseFolderLauncher { uri, path ->
+        val context = context ?: AcraApplication.context ?: return@getChooseFolderLauncher
+        (path ?: uri.toString()).let {
+            PreferenceManager.getDefaultSharedPreferences(context).edit()
+                .putString(getString(R.string.backup_path_key), uri.toString())
+                .putString(getString(R.string.backup_dir_key), it)
+                .apply()
+        }
+    }
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         hideKeyboard()
         setPreferencesFromResource(R.xml.settings_updates, rootKey)
-        //val settingsManager = PreferenceManager.getDefaultSharedPreferences(requireContext())
+        val settingsManager = PreferenceManager.getDefaultSharedPreferences(requireContext())
```

Comment:
using this top level declaration instead of in each listener. any reason to use separate calls?

Reply:
Does it? if thats the case we could have a null check usage of `it.context`.

Reply:
No, you misunderstand.

`requireContext()`: can crash, but no wet code
`it.context`: can't crash, but wet code